src/users/index.ts

```diff
@@ -18,35 +47,202 @@ export interface Users {
    * @returns {Promise<Record<string, Role>>} A map of role names to their respective roles.
    */
   getAssignedRoles: (userId: string) => Promise<Record<string, Role>>;
+
+  db: DBUsers;
+  oidc: OIDCUsers;
+}
+
+/** Operations supported for namespaced 'db' users.*/
+export interface DBUsers extends UsersBase {
   /**
-   * Assign roles to a user.
+   * Retrieve the roles assigned to a 'db_user' user.
    *
-   * @param {string | string[]} roleNames The name or names of the roles to assign.
-   * @param {string} userId The ID of the user to assign the roles to.
-   * @returns {Promise<void>} A promise that resolves when the roles are assigned.
+   * @param {string} userId The ID of the user to retrieve the assigned roles for.
+   * @returns {Promise<Record<string, Role>>} A map of role names to their respective roles.
    */
-  assignRoles: (roleNames: string | string[], userId: string) => Promise<void>;
+  getAssignedRoles: (userId: string, opts?: GetAssignedRolesOptions) => Promise<Record<string, Role>>;
+
+  /** Create a new 'db_user' user.
+   *
+   * @param {string} userId The ID of the user to create. Must consist of valid URL characters only.
+   * @returns {Promise<string>} API key for the newly created user.
+   */
+  create: (userId: string) => Promise<string>;
+
   /**
-   * Revoke roles from a user.
+   * Delete a 'db_user' user. It is not possible to delete 'db_env_user' users programmatically.
    *
-   * @param {string | string[]} roleNames The name or names of the roles to revoke.
-   * @param {string} userId The ID of the user to revoke the roles from.
-   * @returns {Promise<void>} A promise that resolves when the roles are revoked.
+   * @param {string} userId The ID of the user to delete.
+   * @returns {Promise<boolean>} `true` if the user has been successfully deleted.
    */
-  revokeRoles: (roleNames: string | string[], userId: string) => Promise<void>;
+  delete: (userId: string) => Promise<boolean>;
+
+  /**
+   * Rotate the API key of a 'db_user' user. The old API key becomes invalid.
+   * API keys of 'db_env_user' users are defined in the server's environment
+   * and cannot be modified programmatically.
+   *
+   * @param {string} userId The ID of the user to create a new API key for.
+   * @returns {Promise<string>} New API key for the user.
+   */
+  rotateKey: (userId: string) => Promise<string>;
+
+  /**
+   * Activate 'db_user' user.
+   *
+   * @param {string} userId The ID of the user to activate.
+   * @returns {Promise<boolean>} `true` if the user has been successfully activated.
+   */
+  activate: (userId: string) => Promise<boolean>;
+
+  /**
+   * Deactivate 'db_user' user.
+   *
+   * @param {string} userId The ID of the user to deactivate.
+   * @returns {Promise<boolean>} `true` if the user has been successfully deactivated.
+   */
+  deactivate: (userId: string) => Promise<boolean>;
```

Comment:
There is a revoke_key option that is missing (I also forgot it in the python client)

Reply:
`revokeKey` in TS plz 😜